typedspark/_utils/load_table.py

```diff
@@ -35,26 +48,50 @@ def _create_schema(structtype: SparkStructType, schema_name: Optional[str] = Non
     return schema  # type: ignore
 
 
-def _extract_data_type(dtype: DataType) -> Type[DataType]:
+def _extract_data_type(dtype: DataType, name: str) -> Type[DataType]:
     """Given an instance of a ``DataType``, it extracts the corresponding
     ``DataType`` class, potentially including annotations (e.g.
     ``ArrayType[StringType]``)."""
     if isinstance(dtype, SparkArrayType):
-        element_type = _extract_data_type(dtype.elementType)
+        element_type = _extract_data_type(dtype.elementType, name)
         return ArrayType[element_type]  # type: ignore
 
     if isinstance(dtype, SparkMapType):
-        key_type = _extract_data_type(dtype.keyType)
-        value_type = _extract_data_type(dtype.valueType)
+        key_type = _extract_data_type(dtype.keyType, name)
+        value_type = _extract_data_type(dtype.valueType, name)
         return MapType[key_type, value_type]  # type: ignore
 
     if isinstance(dtype, SparkStructType):
-        subschema = _create_schema(dtype)
+        subschema = _create_schema(dtype, _to_camel_case(name))
```

Comment:
Why does it need to be CamelCase here?

Reply:
You're setting the name of a Schema class here. And class names should be camel case.

For example. suppose you're loading:
```python
class B(Schema):
    value_container: Column[StructType[A]]
```

Then the class name for `A` would be `ValueContainer`. It would be really weird to me if it was:
```python
class value_container(Schema):
    a: Column[StringType]
    b: Column[IntegerType]
    c: Column[StringType]
```